### PR TITLE
Update architecture.md

### PR DESCRIPTION
**Pull request by okwme**
Closes: #151 

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -1,64 +1,85 @@
...
+- Prophecy
+    - A data type that represents the first Claim made on the contract about an event coming from an SDK chain. This Claim generates a new Prophecy that has a unique ID used as a reference for subsequent Claims made to support the initial Prophecy.
+- Valset
+    - A contract that keeps track of a set of SDK chain validators and their corresponding token weights.
+- Valset Operator
+    - An actor with the permission to update the validator set within the `Valset` contract.
+    - This is a point of centralization that is addressed in Peggy v3.
+- CosmosBridge
+    - A contract which receives new Prophecies from validators, concludes Prophecies after a threshold of Claims have been made against them within the `Oracle` contract and issues wrapped tokens on behalf of the `BridgeBank` or unlocks tokens previously in escrow.
```
Comment (jkbrooks) [inline docs/architecture.md:16]: Are we open to converging on "pegged token" as the default term?

File: docs/architecture.md [hunk 2]
```diff
@@ -1,64 +1,85 @@
...
+
+### SDK → EVM
+
+In order to move an SDK based Coin to the EVM chain a user would need to sign and execute a `MsgLock` on the `ethbridge` module. This module will move the coins into escrow controlled by that module and event an event called `MsgLock`.
+
+The relayer that is configured to listen for events on the SDK chain will hear the `MsgLock` event and creat a `ProphecyClaim` containing the Lock event destined for the EVM chain. The event would contain the token address of a `BridgeToken` which was created with the same `symbol` as the Coin `denom`. Without a corresponding `BridgeToken` the `ProphecyClaim` will eventually fail.
```
Comment (jkbrooks) [inline docs/architecture.md:79]: *create*

Comment (okwme) [inline docs/architecture.md:79]:
```suggestion
The relayer that is configured to listen for events on the SDK chain will hear the `MsgLock` event and create a `ProphecyClaim` containing the Lock event destined for the EVM chain. The event would contain the token address of a `BridgeToken` which was created with the same `symbol` as the Coin `denom`. Without a corresponding `BridgeToken` the `ProphecyClaim` will eventually fail.

```

File: docs/architecture.md (hunk 1 above)
Comment (okwme) [inline docs/architecture.md:16]:
```suggestion
    - A contract which receives new Prophecies from validators, concludes Prophecies after a threshold of Claims have been made against them within the `Oracle` contract and issues pegged tokens on behalf of the `BridgeBank` or unlocks tokens previously in escrow.

```

**Comment by okwme** [comment]
I think this architecture is still out of date with the changes to whitelist as well, can anyone confirm?

**Comment by denalimarsh** [comment]
> I think this architecture is still out of date with the changes to whitelist as well, can anyone confirm?

In the User Flows > Initialization section the following text should be removed: "When moving Coins from an SDK chain to an EVM chain, the operator of the BridgeToken must first have initiated a new BridgeToken by executing the createNewBridgeToken() function."

Without the whitelist, bridge tokens are automatically deployed for Cosmos assets.

File: docs/architecture.md [hunk 3]
```diff
@@ -1,64 +1,89 @@
...
 
-## The Relayer
+When moving ERC-20 Tokens on an EVM chain to the Peggy equipped SDK chain it should be noted that prior to the transfer an `approve()` must be called on the ERC-20 to allow the `BridgeBank` contract to move tokens on behalf of the user.
 
-The Relayer is a service which interfaces with both blockchains, allowing validators to attest on the Cosmos blockchain that specific events on the Ethereum blockchain have occurred. Through the Relayer service, validators witness the events and submit proofs in the form of signed hashes to the Cosmos based modules, which are responsible for aggregating and tallying the Validators’ signatures and their respective signing power.
+When moving Coins from an SDK chain to an EVM chain, the operator of the `BridgeToken`  must first have initiated a new `BridgeToken` by executing the `createNewBridgeToken()` function.
```
Comment (denalimarsh) [inline docs/architecture.md:60]:
```suggestion

```

**Pull request closed without merging**